reindexer/reindex_request_processor/src/test/scala/uk/ac/wellcome/platform/reindex/processor/services/ReindexWorkerServiceTest.scala

```diff
@@ -0,0 +1,184 @@
+package uk.ac.wellcome.platform.reindex.processor.services
+
+import org.scalatest.FunSpec
+import org.scalatest.concurrent.ScalaFutures
+import uk.ac.wellcome.messaging.sns.NotificationMessage
+import uk.ac.wellcome.messaging.sqs.{SQSConfig, SQSStream}
+import uk.ac.wellcome.messaging.test.fixtures.SQS.{Queue, QueuePair}
+import uk.ac.wellcome.messaging.test.fixtures.{Messaging, SQS}
+import uk.ac.wellcome.models.reindexer.{ReindexRequest, ReindexableRecord}
+import uk.ac.wellcome.monitoring.test.fixtures.MetricsSenderFixture
+import uk.ac.wellcome.storage.test.fixtures.LocalDynamoDb.Table
+import uk.ac.wellcome.storage.test.fixtures.LocalVersionedHybridStore
+import uk.ac.wellcome.storage.test.fixtures.S3.Bucket
+import uk.ac.wellcome.storage.vhs.EmptyMetadata
+import uk.ac.wellcome.test.fixtures.{Akka, TestWith}
+import uk.ac.wellcome.test.utils.ExtendedPatience
+import uk.ac.wellcome.utils.JsonUtil._
+
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class ReindexWorkerServiceTest
+    extends FunSpec
+    with Akka
+    with LocalVersionedHybridStore
+    with SQS
+    with ScalaFutures
+    with Messaging
+    with MetricsSenderFixture
+    with ExtendedPatience {
+
+  val id = "sierra/1234567"
+
+  it("creates a new record") {
+    withLocalSqsQueue { queue =>
+      withLocalS3Bucket { vhsBucket =>
+        withLocalDynamoDbTable { vhsTable =>
+          withReindexWorkerService(vhsTable, vhsBucket, queue) { _ =>
+            val version = 1
```

Comment:
nit: let's use `reindexVersion` or `desiredVersion` to distinguish it from the internal version

also using `Instant.now` / the int equivalent will give you values that look more like the real reindex versions